grid/chat/get-messages/maps/slack.suma

```diff
@@ -0,0 +1,106 @@
+profile = "chat/get-messages@1.0"
+provider = "slack"
+
+"""
+GetMessages map
+"""
+map GetMessages {
+	http GET "/conversations.history" {
+		request "application/x-www-form-urlencoded" {
+			query {
+				channel = input.destination
+				oldest = input.afterDate
+				cursor = input.page
+				limit = input.limit || 100
+			}
+
+			headers {
+				"Authorization" = `Bearer ${parameters.accessToken}`
+			}
+		}
+
+		response 200 "application/json" {
+			// See https://api.slack.com/methods/conversations.history#errors for more information about errors
+			return map error if (body.ok === false) {
+				title = body.error
+			}
+
+			messages = call MapMessages(messages = body.messages)
+
+			return map result if (body.has_more) {
+				nextPage = body.response_metadata.next_cursor
+				messages = messages
+			}
+
+			return map result if (!body.has_more) {
+				messages = messages
+			}
+		}
+
+		response 429 {
+			map error {
+				title = 'Too Many Requests'
+				detail = `Retry after ${headers['Retry-After']} seconds`
+			}
+		}
+
+		response "application/json" {
+			map error if (body.ok === false) {
+				title = body.error
+			}
+		}
+	}
+}
+
+operation MapMessages {
+	messages = args.messages.filter(message => message.type === 'message')
+					
+	set if (messages.length > 0) {
```

Comment:
What do you mean?

Reply:
What I mean is that you could remove that condition completely. Or just chain it into a single call:

```js
args.messages.filter(message => message.type === 'message').map(() => ...)
```

Though more efficient approach to avoid 2 separate iterations would be to do filtering in single pass, e.g.

```js
messages = []
args.messages.forEach((message) => {
  if (message.type !== 'message') {
    return
  }
  // ...
  messages.push(result)
}
```

Reply:
That looks good, I'll refactor it.

Reply:
Unfortunately jessie expression such as `args.messages.forEach` can't be used outside assignment, so I had to chain it, but when I chaned it, language server had problem with that, so I just removed the condition.

Reply:
Ok it no longer throws that error, so I chained it.